ipn/ipnlocal/local.go

```diff
@@ -2358,7 +2360,16 @@ func (b *LocalBackend) enterState(newState ipn.State) {
 		if authURL == "" {
 			systemd.Status("Stopped; run 'tailscale up' to log in")
 		}
-	case ipn.Starting, ipn.NeedsMachineAuth:
+	case ipn.Starting:
+		if oldState == ipn.NeedsLogin && !keyExpired && blocked {
```

Comment:
This all seems oddly specific.

Seems like there might be a bunch of ways to get stuck and we're addressing just one we saw.

Why is key expiry only sticking on NeedsLogin => Starting?

Why don't we unblock in LocalBackend.setClientStatus instead where it's already unblocked?

I mean, how do we know the `blocked` here being true means that it was blocked due to the key being expired?

Reply:
> Why don't we unblock in LocalBackend.setClientStatus instead where it's already unblocked?

That makes a lot more sense, how's this?